### PR TITLE
Incorporate DSCPython3 branch updated to .NET 4.8

### PR DESCRIPTION
**Pull request by mmisol**
### Purpose

Changes in this PR:
1. Includes DSCPython3 branch with merged changes from latest master
2. Upgrades projects and NuGet packages to .NET 4.8

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoApplications/PathResolvers.cs [hunk 1]
```diff
@@ -30,6 +30,7 @@ public SandboxPathResolver(string preloaderLocation)
                 "DSCoreNodes.dll",
                 "DSOffice.dll",
                 "DSIronPython.dll",
+                "DSCPython.dll",
```
Comment (mjkkirschner) [inline src/DynamoApplications/PathResolvers.cs:33]: I guess this should also be added to the CLI path resolver?

Comment (mmisol) [reply src/DynamoApplications/PathResolvers.cs:33]: Thanks @mjkkirschner , I will update that. Let me know if you find anything else.

File: src/Libraries/DSCPython/CPythonEvaluator.cs [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using Autodesk.DesignScript.Runtime;
+using Dynamo.Utilities;
+using Python.Runtime;
+
+namespace DSCPython
+{
+    [SupressImportIntoVM]
+    public enum EvaluationState { Begin, Success, Failed }
```
Comment (mjkkirschner) [inline src/Libraries/DSCPython/CPythonEvaluator.cs:12]: if this needs to be public can we add summary tags?
If not can we make it internal?

File: src/Libraries/DSCPython/CPythonEvaluator.cs [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+    [SupressImportIntoVM]
+    public delegate void EvaluationEventHandler(EvaluationState state,
```
Comment (mjkkirschner) [inline src/Libraries/DSCPython/CPythonEvaluator.cs:15]: same here(public/internal)

File: src/Libraries/DSCPython/CPythonEvaluator.cs [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Autodesk.DesignScript.Runtime;
+using Dynamo.Utilities;
+using Python.Runtime;
+
+namespace DSCPython
+{
+    [SupressImportIntoVM]
```
Comment (mjkkirschner) [inline src/Libraries/DSCPython/CPythonEvaluator.cs:11]: is this class actually imported into the VM at all?

Comment (mmisol) [reply src/Libraries/DSCPython/CPythonEvaluator.cs:11]: I'm not sure about this. Maybe @reddyashish ?

**Comment by mmisol** [comment]
I'll go ahead and merge this. Hopefully the green build will return soon.

File: src/Libraries/DSCPython/CPythonEvaluator.cs (hunk 4 above)
Comment (reddyashish) [reply src/Libraries/DSCPython/CPythonEvaluator.cs:11]: Yes, the class is imported in the VM but not these internal variables. Similar to what is done in IronPythonEvaluator class.

**Pull request merged**